### PR TITLE
[inductor] Support IndexPutFallback in cpp_wrapper

### PR DESCRIPTION
**Pull request by desertfire**
Summary:
1) Make the fallback index_put generate the right cpp code in cpp_wapper
2) Add a --cpp-wrapper option to common.py

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -2939,10 +2934,11 @@ def codegen(self, wrapper):
             if self.indices[i] is not None:
                 indices.append(next(iter_valid_indices))
             else:
-                indices.append("None")
-        wrapper.writeline(
-            f"{self.kernel}({x}, [{','.join(indices)}], {values}, {repr(self.constant_args[0])})"
-        )
+                indices.append(V.graph.wrapper_code.none_str)
+
+        indices = f"{V.graph.wrapper_code.open_bracket}{', '.join(indices)}{V.graph.wrapper_code.closed_bracket}"
```
Comment (jgong5) [inline torch/_inductor/ir.py:2939]: How about add a function to make it less verbose, e.g., 
```suggestion
        indices = f"{V.graph.wrapper_code.codegen_list_init(', '.join(indices))}"
```

**Review by jgong5 — APPROVED** [review]
LGTM. Check a minor suggested edit.

**Review by jansel — APPROVED** [review]

File: torch/_inductor/ir.py (hunk 1 above)
Comment (desertfire) [reply torch/_inductor/ir.py:2939]: I will do it in a later PR.

**Pull request closed without merging**